pandas/tests/groupby/test_grouping.py

```diff
@@ -251,6 +251,37 @@ def test_groupby_levels_and_columns(self):
         by_columns.columns = pd.Index(by_columns.columns, dtype=np.int64)
         tm.assert_frame_equal(by_levels, by_columns)
 
+    def test_groupby_categorical_index_and_columns(self):
+        # GH18432
+        columns = ['A', 'B', 'A', 'B']
+        categories = ['B', 'A']
+        cat_columns = CategoricalIndex(columns,
+                                       categories=categories,
```

Comment:
can you change this test to instead of using ranging data, uses integer as the data, then directly construct the result frame

Reply:
Yes. I made the change. I think it is what you had in mind and I agree it is more straightforward this way.